spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -146,13 +148,29 @@ public RewriteDataFilesSparkAction filter(Expression expression) {
     return this;
   }
 
+  @Override
+  public RewriteDataFiles targetBranch(String branch) {
+    this.targetBranch = branch;
+    return this;
+  }
+
+  protected long startingSnapshotId() {
+    if (SnapshotRef.MAIN_BRANCH.equals(this.targetBranch)) {
+      return table.currentSnapshot().snapshotId();
+    } else {
+      SnapshotRef ref = table.refs().get(this.targetBranch);
+      assert (ref != null);
```

Comment:
we typically don't use `assert` because they need to be enabled. So what makes more sense is to use `Preconditions.checkArgument(...)`